### ISSUE TITLE
file attributes should be applied regardless of file contents

### ISSUE THREAD

**Issue by mmoya** [ansible/ansible#2140]
Typical template task:

```
- name: test template
  template: templates/apt.conf.local.j2
            dest=/etc/apt/apt.conf.d/99local
            mode=0644 owner=deploy group=deploy
```

`template` module only acts when content differs. In this case changes to `mode`, `owner` or `group` are ignored if dest has the expected content.

Steps to reproduce:
- Run the above task when `dest` file is not present in remote system.
- In the remote system change any of the destination's mode, owner and group.
- Re-run the above task.
  - As the content has not changed, the module doesn't restore mode, owner and group.
- Alter destination's content.
- Re-run the above task.
  - The content has changed, the module restores content, mode, owner and group.

This is highPriority in my opinion.

**Comment by mpdehaan** [comment]
Indeed it is.   This is easy to fix, thanks for the heads up.

**Comment by mpdehaan** [comment]
Fixed in 6d604469d8cbdaf0bdf5915b12f5d1f4bf07f62f
